src/connectors/hits-per-page/connectHitsPerPage.js

```diff
@@ -118,10 +118,17 @@ export default function connectHitsPerPage(renderFn) {
 
     return {
       getConfiguration() {
-        const { value: defaultHitsPerPage } =
-          items.find(item => item.default) || items[0];
+        if (items.filter(item => item.default).length > 1 && window.console) {
+          window.console.warn(
+            `[Warning][hitsPerPageSelector] more than one default value is specified in \`items[]\`
+  The first one will be picked, you should probably set only one default value`
+          );
+        }
 
-        return { hitsPerPage: defaultHitsPerPage };
+        const defaultHitsPerPage = items.find(item => item.default);
```

Comment:
That's not big but there are 2 loops over items (one with filter and the other with find). We could store the result of the filter and reuse it here.